lib/authenticated-api/authenticated-api.ts

```diff
@@ -59,17 +62,56 @@ export class AuthenticatedApi extends cdk.Construct {
       }),
     };
 
-    this.httpApi = new apigatewayv2.HttpApi(
-      this,
-      `${props.prefix}${props.name}`,
-      apiGatewayProps
-    );
+    this.httpApi = new apigatewayv2.HttpApi(this, apiName, apiGatewayProps);
 
     this.apiId = this.httpApi.apiId;
     this.httpApiId = this.httpApi.httpApiId;
 
+    if (props.logging?.enabled) {
+      const logGroup = new awslogs.LogGroup(this, "apiGatewayLogs", {
+        // CloudWatch Logs automatically enables the /aws/vendedlogs/* prefix
+        // in the resource policy, so we use it to avoid adding each log
+        // group to the policy, which has a limit of 5120 characters.
+        logGroupName: `/aws/vendedlogs/${apiName}-accessLog`,
```

Comment:
When you enable logging on APIGateway, AWS automatically tries to put your resource's ARN into a resource policy called "AWSLogDeliveryWrite20150319". Trying to deploy a new stack, I got this error:

> Cannot enable logging. Policy document length breaking Cloudwatch Logs Constraints, either < 1 or > 5120

The [documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/AWS-logs-and-resource-policy.html#AWS-logs-infrastructure-CWL) explains this:

> ### Log group resource policy size limit considerations
>
> These services must list each log group that they're sending logs to in the resource policy, and CloudWatch Logs resource policies are limited to 5120 characters. A service that sends logs to a large number of log groups may run into this limit.
>
> To mitigate this, CloudWatch Logs monitors the size of resource policies used by the service that is sending logs, and when it detects that a policy approaches the size limit of 5120 characters, CloudWatch Logs automatically enables /aws/vendedlogs/* in the resource policy for that service. You can then start using log groups with names that start with /aws/vendedlogs/ as the destinations for logs from these services.

So in order to avoid having issues the policy again, it's best to use the `/aws/vendedlogs/` prefix for the log group name.

Our policy was full of build and development resources, so I edited it to remove them.
Resource policies cannot be edited in AWS Console, but can be seen in the CLI with:

```sh
aws logs describe-resource-policies

# to see policy JSON:
aws logs describe-resource-policies --query 'resourcePolicies[?policyName==`AWSLogDeliveryWrite20150319`].policyDocument' --output json | jq '.[0] | fromjson'

# to update the policy:
aws logs put-resource-policy --policy-name AWSLogDeliveryWrite20150319 --policy-document "<policy JSON>"
```

There is also an option to manage that policy with Terraform via [`api_gateway_account`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_account) resource.

Reply:
Was this the reason we backed this out after it was added in the first time?

Reply:
It's hard to tell - CircleCI pipeline logs expire and I couldn't see the exact issue we had then, but it's quite possible that was it.